label_trans_selinux.go

```diff
@@ -20,6 +18,28 @@ const timeout = time.Second * 3
 
 const nullChar = "\000"
 
+/* This is a cursed function and I hate it.
+ * The Go devlopers have decided that machine byteorder is an unimportant detail
+ * and 1) force you to specify order when decoding byte arrays, and
+ * 2) don't give you a simple way of figuring out the native endianness
+ * so here we go. Create a pointer and then look at the byte order to set
+ * the interface for reading off of the socket. */
+var nativeEndian binary.ByteOrder
+
+func setNativeEndian() {
```

Comment:
If you change this function to have a pointer receiver of Conn, you can add `nativeEndian` to the Conn data structure. If you also make `sendRequest` take a `Conn` pointer receiver, you can access the variable directly instead of having `nativeEndian` be a global. It seems cleaner overall that way.